src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/DocumentPullDiagnosticsHandler.cs

```diff
@@ -31,7 +31,7 @@ internal class DocumentPullDiagnosticsHandler :
             LSPDocumentManager documentManager,
             LSPDocumentSynchronizer documentSynchronizer,
             LSPDiagnosticsTranslator diagnosticsProvider,
-            HTMLCSharpLanguageServerFeedbackFileLoggerProvider loggerProvider)
+            HTMLCSharpLanguageServerLogHubLoggerProvider loggerProvider)
```

Comment:
Maybe not for this PR, but might it make sense to abstract this out a little bit so this kind of change could be "drop-in" in the future?

Reply:
> abstract this out a little bit so this kind of change could be "drop-in" in the future?

Hmm, I'm not sure I follow. Could you please give an example?

Reply:
I mean instead of taking a `HTMLCSharpLanguageServerLogHubLoggerProvider` here does it make more sense if we define `IHTMLCSharpLanguageServerLoggerProvider` (or `ILoggerProvider` depending on how specific these need to be) and just leave the implementation details (is it `LogHub`, `FeedbackFile` or other) up to DI.

Reply:
Normally, yes 💯 . 

In this case however, I think the current implementation may be required. Essentially we use `HTMLCSharpLanguageServerLogHubLoggerProvider` and `RazorLanguageServerLogHubLoggerProvider` to create two separate logger providers for the appropriate language servers (essentially leveraging statically typed objects & DI to ensure we don't create duplicate loggers). There is some minor language server specific functionality, but it mainly goes to a LoggerProviderBase class which itself implements `ILoggerProvider`. We could further abstract to `IHTMLCSharpLanguageServerLogHubLoggerProvider` & `IRazorLanguageServerLogHubLoggerProvider`, but I think at this point that may just create more complexity in this scenario.